File: backend/api/routers/ai.py
from fastapi import APIRouter, HTTPException, status, Depends, UploadFile, File
import requests
from .depends import check_audio_id

router = APIRouter(prefix="/ai", tags=["AI"])

ip = "35.192.107.93"


@router.get("/summarize/{audio_id}", status_code=status.HTTP_200_OK)
def get_summary(audio_id: int = Depends(check_audio_id)):
    response = requests.get("http://" + ip + ":8888/summarize/" + str(audio_id))
    summary = response.json()
    return summary


@router.get("/inference/{audio_id}", status_code=status.HTTP_200_OK)
def inference_audio(audio_id: int = Depends(check_audio_id)):
    response = requests.get("http://" + ip + ":8888/inference/" + str(audio_id))
    inference = response.json()
    return inference


@router.get("/transcript/{audio_id}", status_code=status.HTTP_200_OK)
def get_transcript(audio_id: int = Depends(check_audio_id)):
    response = requests.get("http://" + ip + ":8888/transcript/" + str(audio_id))
    transcript = response.json()
    return transcript


@router.get("/term/{audio_id}", status_code=status.HTTP_200_OK)
def get_term(audio_id: int = Depends(check_audio_id)):
    response = requests.get("http://" + ip + ":8888/term/" + str(audio_id))
    term = response.json()
    return term


@router.get("/chatbot", status_code=status.HTTP_200_OK)
def post_chatbot_q(query: str):
    response = requests.post("http://" + ip + ":8888/chatbot?query=" + query)
    reply = response.json()
    return reply

File: frontend/src/api/ai.ts
import axios from "axios";
import { Summary } from "../schema/summary"
import { Transcript } from "../schema/transcript";
import { Term } from "../schema/term";
import { Reply } from "../schema/reply";

export async function getSummary(audio_id : number):Promise<Summary>{
    let url = "http://127.0.0.1:8080/ai/summarize/" + audio_id;
    let course;
    try {
        const response = await axios.get(url);
        course = response.data;
    }
    catch(error) {  
        
    }
    return course;
}

export async function getTranscript(audio_id : number):Promise<Transcript>{
    let url = "http://127.0.0.1:8080/ai/transcript/" + audio_id;
    let transcript;
    try {
        const response = await axios.get(url);
        transcript = response.data;
    }
    catch(error) {  
        
    }
    return transcript;
}

export async function getTerm(audio_id : number):Promise<Term>{
    let url = "http://127.0.0.1:8080/ai/term/" + audio_id;
    let term;
    try {
        const response = await axios.get(url);
        term = response.data;
    }
    catch(error) {  
        
    }
    return term;
}

export async function doInference(audio_id : number):Promise<void>{
    let url = "http://127.0.0.1:8080/ai/inference/" + audio_id;
    try {
        const response = await axios.get(url);
    }
    catch(error) {  
        
    }
}

export async function getReply(query :string):Promise<Reply>{
    let url = "http://127.0.0.1:8080/ai/chatbot?query=" + query;
    let reply;
    try {
        const response = await axios.get(url);
        reply = response.data;
    }
    catch(error) {  
        
    }
    return reply;
}

